packages/core/lib/command-utils.js

```diff
@@ -107,8 +112,14 @@ const prepareOptions = ({ command, inputStrings, options }) => {
   });
 
   // method `extractFlags(args)` : Extracts the `--option` & `-option` flags from arguments
-  const inputOptions = extractFlags(inputStrings);
+  let inputOptions = extractFlags(inputStrings);
 
+  //prevent invalid option warning for `-v` & `--version`
```

Comment:
One more nit and then this looks good; can you make it clear here that it is for `truffle -v` and `truffle --version`? The comment as it is written still seems like it would be confusing for someone. Other than that, if you have tested this I think we are good to go!